File: Data-Structures-And-Algorithms/Array/leetCode-11-maxArea.go
func maxArea(height []int) int {
	left, right, ans := 0, len(height)-1, 0
	for left < right {
		area := (right - left) * min(height[left], height[right])
		if area > ans {
			ans = area
		}
		if height[left] > height[right] {
			right--
		} else {
			left++
		}
	}
	return ans
}
func min(x, y int) int {
	if x > y {
		return y
	}
	return x
}

File: Data-Structures-And-Algorithms/Array/leetCode-11-maxArea.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1, max = 0;
        while (left < right) {
            max = Math.max(max, (right-left) * Math.min(height[right], height[left]));
            if (height[right] > height[left]) {
                left++;
            } else {
                right--;
            }
        }
        return max;
    }
}

